addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1027,6 +1030,61 @@ options.registry.anchor = options.Class.extend({
     },
 });
 
+options.registry.CookiesBar = options.registry.SnippetPopup.extend({
+    xmlDependencies: [
+        '/website/static/src/xml/website.cookies_bar.xml',
+    ],
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Change the cookies bar layout.
+     *
+     * @see this.selectClass for parameters
+     */
+    selectLayout: function (previewMode, widgetValue, params) {
+        let websiteId;
+        this.trigger_up('context_get', {
+            callback: function (ctx) {
+                websiteId = ctx['website_id'];
+            },
+        });
+
+        const template = $(qweb.render(`website.cookies_bar.${widgetValue}`, {
+            websiteId: websiteId,
+        }));
+
+        const $content = this.$target.find('.s_popup_content');
+        const selectorToKeep = [
+            '.cookies_bar_text_button',
+            '.cookies_bar_text_policy',
+            '.cookies_bar_text_title',
+            '.cookies_bar_text_primary',
+            '.cookies_bar_text_secondary',
+        ];
+
+        if (this.text === undefined) {
+            this.text = [];
```

Comment:
More nitpicking (sorry!): storing markup in `text` variable is a bit confusing, like mixing `$.html()` and `$.text()`. Maybe name it `contentNodes`, or `contentSource`?

BTW, how about having all the content sections present in all 3 layouts, but some of them `d-none` somewhere? Consequences:
- No need to store the contents in JS vars at all, you can directly move the nodes to the new place with `$.appendTo()`. Bonus: no need for extracting `innerHTML` and reinjecting it, so we save one more `$.html` :tada: 
- The hidden contents is preserved for future layout changes as well, in another editor session, rather than lost after save.
- Minor drawback: the user might drop the hidden elements accidentally, so that data won't be there at the next layout change. No big deal, there are other ways to lose contents, including visible stuff. The transfer of contents between layouts is just a best effort thing.

Reply:
On second thought, we end up polluting the DOM with hidden contents that may stay there forever, and that probably sucks. So nevermind, the above. Yet the variable name could still be improved ;-)

Reply:
Yep, we had some debate about this and I was against keeping d-none content in the DOM *unless it was customized by the user at some point and that it makes sense for the user not to lose it*. Our only example for now is our new "restaurant menu"-like snippet which may or not display descriptions for the menu items. On save, we keep the hidden descriptions **that were not customized**, so that the user may re-enable them someday but we remove the ones that were not customized to not pollute the DOM for no reason. Another potential reason, even if probably not the case anymore (the internet is not very clear about it): hidden text could be considered for SEO.

Anyways, I did not check the final result here yet. I'll see if we can do better yet.